api/common/api-profile.go

```diff
@@ -0,0 +1,457 @@
+	case string:
+		valueStr = v
+	case *string:
+		valueStr = *v
+	default:
+		// Should not happen if correctly implemented
+		return cerrors.CreateInternalServerError("unknownInputType")
+	}
+	for _, opt := range validator["options"].([]interface{}) {
+		if valueStr == opt.(string) {
+			return nil
+		}
+	}
+	return cerrors.CreateBadRequestError(cerrors.MsgErrInvalidParam + "." + *attrb.Name)
+}
+
+// Ensure value is not nil before calling this
+func validateAttributeURI(attrb kc.ProfileAttrbRepresentation, validator kc.ProfileAttrValidatorRepresentation, value interface{}) error {
+	switch v := value.(type) {
+	case string:
+		return validateRegexWithString(attrb, constants.RegExpRedirectURI, v)
+	case *string:
+		return validateRegexWithString(attrb, constants.RegExpRedirectURI, *v)
+	default:
+		// Should not happen if correctly implemented
+		return cerrors.CreateInternalServerError("unknownInputType")
+	}
+}
+
+// Ensure value is not nil before calling this
+func validateAttributeLocalDate(attrb kc.ProfileAttrbRepresentation, validator kc.ProfileAttrValidatorRepresentation, value interface{}) error {
+	var ctValidate = validation.NewParameterValidator()
+	switch v := value.(type) {
+	case string:
+		return ctValidate.ValidateParameterDateMultipleLayout(*attrb.Name, &v, constants.SupportedDateLayouts, false).Status()
+	case *string:
+		return ctValidate.ValidateParameterDateMultipleLayout(*attrb.Name, v, constants.SupportedDateLayouts, false).Status()
+	default:
+		// Should not happen if correctly implemented
+		return cerrors.CreateInternalServerError("unknownInputType")
+	}
+}
+
+func validateRegexWithString(attrb kc.ProfileAttrbRepresentation, regex string, value string) error {
+	// Java: \p{L} Golang: \p{Lu}\p{Ll}
+	// Replaces Java regex not recognized by Golang
+	regex = strings.ReplaceAll(regex, `\h`, `\s`)                   // Horizontal spaces
+	regex = strings.ReplaceAll(regex, `\v`, `\s`)                   // Vertical spaces
+	regex = strings.ReplaceAll(regex, `\p{Cntrl}`, `\x00-\x1F\x7F`) // Control characters
+	res, _ := regexp.MatchString(regex, value)
+	if !res {
+		return cerrors.CreateBadRequestError(cerrors.MsgErrInvalidParam + "." + *attrb.Name)
+	}
+	return nil
+}
+
+func validateRangeWithString(attrb kc.ProfileAttrbRepresentation, validator kc.ProfileAttrValidatorRepresentation, referenceValue int) error {
+	for k, v := range validator {
+		if k == "min" {
+			var attributeValue = toInt(v, referenceValue+1) // Use a too high value by default to make the test fails
+			if referenceValue < attributeValue {
+				return cerrors.CreateBadRequestError(cerrors.MsgErrInvalidParam + "." + *attrb.Name)
+			}
+		} else if k == "max" {
+			var attributeValue = toInt(v, referenceValue-1) // Use a too low value by default to make the test fails
+			if referenceValue > attributeValue {
+				return cerrors.CreateBadRequestError(cerrors.MsgErrInvalidParam + "." + *attrb.Name)
+			}
+		}
+	}
+	return nil
+}
+
```

Comment:
Maybe all this generic conversion methods bellow could be move to commons ?
It would allow us to reuse it and reduce the size of this big api-profile, what do you think ?